geochemistrypi/data_mining/process/network.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+import pandas as pd
+
+from ..model.network import BronKerboschWorkflow, LouvainMethodWorkflow
+from ._base import ModelSelectionBase
+
+
+class NetworkAnalysisModelSelection(ModelSelectionBase):
+    def __init__(self, model_name: str) -> None:
+        # Initialize the model name
+        self.model_name = model_name
+        # Initialize the corresponding network analysis workflow based on the model name
```

Comment:
Move this initialization into "def activate". This is the convention. Can refer to process/regress.py.